### PR TITLE
tasknet-5-6/load-pipeline-improvements

### LINKED ISSUE #4: Return all scores from a text classification pipeline
Currently this is not a parameter that we can pass to the `TextClassificationPipeline` from the `load_pipeline` function. It would be interesting to have the functionality when using the model on specific tasks for inference.

### LINKED ISSUE #5: Specify device when loading pipelines
At the moment we cannot specify a device when loading a pipeline using the `load_pipeline` helper function. To have a fine grained control over how pipelines are loaded, it would be nice too pass it to the `TextClassificationPipeline` constructor.

### PR DESCRIPTION
**Pull request by tboquet**
This closes #4 and closes #5 and adresses the comments.

### CODE REVIEW

**Comment by tboquet** [comment]
Sorry my editor fixed the style, the main changes are in the `load_pipeline` function.

File: src/tasknet/utils.py [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 import magicattr
+import torch
+from datasets import Dataset, DatasetDict, concatenate_datasets, load_dataset
+from easydict import EasyDict as edict
+from tqdm.auto import tqdm
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoTokenizer,
+    TextClassificationPipeline,
+)
```
Comment (tboquet) [inline src/tasknet/utils.py:14]: Is it a problem to have the imports at the module level rather than at the `load_pipeline` level?

**Comment by sileod** [comment]
Thanks a lot, I will inspect that

File: src/tasknet/utils.py [hunk 2]
```diff
@@ -107,57 +128,95 @@ def nested_children(m: torch.nn.Module):
...
-def load_pipeline(model_name, task_name, adapt_task_embedding=True,multilingual=False):
-    if multilingual or 'mdeberta' in model_name:
-        multilingual=True 
+def load_pipeline(
+    model_name: str,
+    task_name: str,
+    adapt_task_embedding: bool = True,
+    multilingual: bool = False,
+    device: int = -1,
+    return_all_scores: bool = False,
```
Comment (tboquet) [inline src/tasknet/utils.py:166]: This is the only real change made here.

File: src/tasknet/utils.py [hunk 3]
```diff
@@ -107,57 +128,95 @@ def nested_children(m: torch.nn.Module):
...
+            ] += adapter.Z[task_index]
 
     pipe = TextClassificationPipeline(
-    model=model, tokenizer=tokenizer)
-    return pipe
\ No newline at end of file
+        model=model,
+        tokenizer=tokenizer,
+        device=device,
+        return_all_scores=return_all_scores,
```
Comment (tboquet) [inline src/tasknet/utils.py:220]: This is the only real change made here.

**Pull request merged**